### PR TITLE
Adds initial router, routes, and placeholder views

### LINKED ISSUE #14802: Add routing + scaffolding of views

### PR DESCRIPTION
**Pull request by ryanahamilton**
Resolves #14802.

This adds `react-router-dom` with an initial collection of Routes and corresponding placeholder view components.

### CODE REVIEW

**Comment by bbovenzi** [comment]
Looks good! But I think the test will break. Could we update it to test the router (ie our main page will render on `/` and the 404 page will show up on `/invalid-path`?

File: airflow/ui/src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,61 @@
...
+import Docs from 'views/Docs';
+import NotFound from 'views/NotFound';
 
 const App = () => (
-  <Center height="100vh">
-    <Heading>Apache Airflow new UI</Heading>
-  </Center>
+  <Switch>
+    <Redirect exact path="/" to="/pipelines" />
+    <Route exact path="/pipelines" component={Pipelines} />
```
Comment (ashb) [inline airflow/ui/src/App.tsx:41]:
```suggestion
    <Route exact path="/workflows" component={Workflows} />
```

etc?

Most of the place in Airflow docs we call it DAGs (most commonly) or Workflows less frequently. Pipeline is very rarely used.

File: airflow/ui/src/App.tsx [hunk 2]
```diff
@@ -1,11 +1,61 @@
...
+    <Route exact path="/config" component={Config} />
+
+    <Route exact path="/access" component={Access} />
+    <Route exact path="/access/users" component={Users} />
+    <Route exact path="/access/users/new" component={Users} />
+    <Route exact path="/access/users/:id" component={Users} />
+    <Route exact path="/access/users/:id/edit" component={Users} />
+    <Route exact path="/access/roles" component={Roles} />
+
+    <Route exact path="/docs" component={Docs} />
```
Comment (ashb) [inline airflow/ui/src/App.tsx:55]: This was previously just a link out to docs -- do we need/want it to be a component now?

File: airflow/ui/src/App.tsx (hunk 1 above)
Comment (ryanahamilton) [reply airflow/ui/src/App.tsx:41]: These are just placeholders, nothing is set in stone. Once we have a full picture of what the new information architecture looks like, we'll be seeking feedback.

File: airflow/ui/src/App.tsx (hunk 2 above)
Comment (ryanahamilton) [reply airflow/ui/src/App.tsx:55]: Yes. Again, nothing set in stone, but we're thinking we can provide a better experience with [a page like we mocked up in the POC](https://nova.astronomer.io/docs).

**Comment by bbovenzi** [comment]
- Updated the tests to properly test routing.
- Fixed `yarn lint` by stopping it from doing typescript checks on `node_modules` packages
- Workaround to prevent the `motion()` warning when running tests

File: airflow/ui/package.json [hunk 3]
```diff
@@ -42,5 +42,8 @@
     "webpack": "^4",
     "webpack-cli": "^3",
     "webpack-dev-server": "^3"
+  },
+  "resolutions": {
```
Comment (ashb) [inline airflow/ui/package.json]: What's this key for? Not seen this one before.

**Review by ashb — APPROVED** [review]

File: airflow/ui/package.json (hunk 3 above)
Comment (bbovenzi) [reply airflow/ui/package.json]: A warning was bubbling up from Chakra UI. They're using a deprecated function from a dependency. `resolutions` let's us swap out their dependency to a slightly older one which doesn't fire the warning. We can remove it when Chakra releases an update.

See:
https://github.com/chakra-ui/chakra-ui/issues/3459#issuecomment-792906739
https://classic.yarnpkg.com/en/docs/selective-version-resolutions/

Comment (ashb) [reply airflow/ui/package.json]: Ohh yeah yarn fix for updating nested dependencies. Gotcha.

We can keep this if it's helpful.

Comment (bbovenzi) [reply airflow/ui/package.json]: Yarn was erroring in the CI build so we're seeing if this caused it.

**Comment by ryanahamilton** [comment]
Closing in favor of #14927

**Pull request closed without merging**